src/backend/renderer/multigpu/mod.rs

```diff
@@ -618,74 +618,90 @@ impl<'a, 'b, R: GraphicsApi, T: GraphicsApi, Target> AsMut<<R::Device as ApiDevi
     }
 }
 
-// Hack for implementing Renderer::render..
-#[derive(Debug)]
-enum RenderDevice<'a, A: GraphicsApi> {
-    Device(&'a mut A::Device),
-    // Hack to avoid lifetime problems in Renderer::render
-    Renderer(*mut <A::Device as ApiDevice>::Renderer, DrmNode),
-}
-
-impl<'a, A: GraphicsApi> RenderDevice<'a, A> {
-    /*
-    fn unwrap_device(&mut self) -> &mut A::Device {
-        match self {
-            RenderDevice::Device(dev) => *dev,
-            RenderDevice::Renderer(_, _) => panic!("unwrap called on RenderDevice::Renderer"),
-        }
-    }
-    */
-
-    fn node(&self) -> &DrmNode {
-        match self {
-            RenderDevice::Device(dev) => dev.node(),
-            RenderDevice::Renderer(_, node) => node,
-        }
-    }
-
-    fn renderer(&self) -> &<A::Device as ApiDevice>::Renderer {
-        match self {
-            RenderDevice::Device(dev) => dev.renderer(),
-            RenderDevice::Renderer(renderer, _) => unsafe { &**renderer },
-        }
-    }
-
-    fn renderer_mut(&mut self) -> &mut <A::Device as ApiDevice>::Renderer {
-        match self {
-            RenderDevice::Device(dev) => dev.renderer_mut(),
-            RenderDevice::Renderer(renderer, _) => unsafe { &mut **renderer },
-        }
-    }
-}
-
 /// [`Frame`](super::Frame) implementation of a [`MultiRenderer`].
-#[derive(Debug)]
-pub struct MultiFrame<R: GraphicsApi, T: GraphicsApi> {
+pub struct MultiFrame<'a, 'b, 'c, R: GraphicsApi + 'c, T: GraphicsApi, Target>
```

Comment:
These lifetime names are meaningless to me.

Reply:
Changed into `'render, 'target, 'frame` for the whole module.